sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -267,3 +256,193 @@ def decode_to_text(encoding, content):
     if encoding:
         return content.decode(encoding)
     return codecs.getincrementaldecoder("utf-8-sig")(errors="replace").decode(content)
+
+class HttpRequestBackcompatMixin(object):
+
+    def __getattr__(self, attr):
+        backcompat_attrs = [
+            "files",
+            "data",
+            "multipart_mixed_info",
+            "query",
+            "body",
+            "format_parameters",
+            "set_streamed_data_body",
+            "set_text_body",
+            "set_xml_body",
+            "set_json_body",
+            "set_formdata_body",
+            "set_bytes_body",
+            "set_multipart_mixed",
+            "prepare_multipart_body",
+            "serialize",
+        ]
+        attr = _pad_attr_name(attr, backcompat_attrs)
+        return self.__getattribute__(attr)
+
+    def __setattr__(self, attr, value):
+        backcompat_attrs = [
+            "multipart_mixed_info",
+            "files",
+            "data",
+            "body",
+        ]
+        attr = _pad_attr_name(attr, backcompat_attrs)
+        super(HttpRequestBackcompatMixin, self).__setattr__(attr, value)
+
+    @property
+    def _multipart_mixed_info(self):
```

Comment:
making all of these backcompat properties hidden, then overriding `getattr` to return them. This ensures that they don't show up as public, even with dir, and they don't get included in sphinx documentation. 
[Here](https://docsupport.blob.core.windows.net/$web/azure-core/azure.core.html#azure.core.rest.HttpRequest) are the generated docs for this request

Thank you @annatisch and @johanste for the useful hack!